llvm/lib/Analysis/InlineAdvisor.cpp

```diff
@@ -63,6 +66,50 @@ static cl::opt<bool>
 
 extern cl::opt<InlinerFunctionImportStatsOpts> InlinerFunctionImportStats;
 
+/// Class to handle dynamic inliner
+class DLInlineAdvisorInfo {
```

Comment:
does it make sense to convert this to a single function "loadDynamicAdvisor(StringRef Path, Module &M, FunctionAnalysisManager &FAM, LLVMContext &Context, ...)?